test/index.ts

```diff
@@ -983,49 +1027,76 @@ schema = Joi.link(str);
-        });
-        let result: Joi.ValidationResult;
-        let asyncResult: Promise<TResult>;
-
-        result = schema.validate(value);
-        if (result.error) {
-            throw Error('error should not be set');
-        } else {
-            expect.type<TResult>(result.value);
-        }
-        result = schema.validate(value, validOpts);
-        asyncResult = schema.validateAsync(value);
-        asyncResult = schema.validateAsync(value, validOpts);
-
-        asyncResult
-            .then(val => JSON.stringify(val, null, 2))
-            .then(val => { throw new Error('one error'); })
-            .catch(e => { });
-
-        expect.type<Promise<TResult>>(schema.validateAsync(value));
-        expect.type<Promise<{ value: TResult, artifacts: Map<any, string[][]> }>>(schema.validateAsync(value, { artifacts: true }));
-        expect.type<Promise<{ value: TResult, warning: Joi.ValidationWarning }>>(schema.validateAsync(value, { warnings: true }));
-        expect.type<Promise<{ value: TResult, artifacts: Map<any, string[][]>; warning: Joi.ValidationWarning }>>(schema.validateAsync(value, { artifacts: true, warnings: true }));
-        expect.error<Promise<{ value: TResult, warning: Joi.ValidationWarning }>>(schema.validateAsync(value, { artifacts: true }));
-        expect.error<Promise<{ value: TResult, artifacts: Map<any, string[][]> }>>(schema.validateAsync(value, { warnings: true }));
-        expect.error<Promise<TResult>>(schema.validateAsync(value, { artifacts: true, warnings: true }));
-        expect.type<Promise<TResult>>(schema.validateAsync(value, { artifacts: false }));
-        expect.type<Promise<TResult>>(schema.validateAsync(value, { warnings: false }));
-        expect.type<Promise<TResult>>(schema.validateAsync(value, { artifacts: false, warnings: false }));
-
-        const falsyValue = { username: 'example' };
-        result = schema.validate(falsyValue);
-        if (!result.error) {
-            throw Error('error should be set');
-        }
+{
+  // validate tests
+  {
+    let value = { username: 'example', password: 'example' };
+    type TResult = { username: string; password: string };
+    const schema = Joi.object<TResult>().keys({
+      username: Joi.string().max(255).required(),
+      password: Joi.string()
+        .pattern(/^[a-zA-Z0-9]{3,255}$/)
+        .required(),
+    });
+    let result: Joi.ValidationResult;
+    let asyncResult: Promise<TResult>;
+
+    result = schema.validate(value);
+    if (result.error) {
+      throw Error('error should not be set');
+    } else {
+      expect.type<TResult>(result.value);
+    }
+    result = schema.validate(value, validOpts);
+    asyncResult = schema.validateAsync(value);
+    asyncResult = schema.validateAsync(value, validOpts);
+
+    asyncResult
+      .then((val) => JSON.stringify(val, null, 2))
+      .then((val) => {
+        throw new Error('one error');
+      })
+      .catch((e) => {});
```

Comment:
What exactly is this supposed to be doing?

Reply:
That's a good question, I just reformatted that file and didn't pay attention to this, it seems to have been there from the import of DT types. I have no idea what it's for as it doesn't appear to be testing types.